lib/session/messageSession.ts

```diff
@@ -625,7 +625,7 @@ export class MessageSession extends LinkEntity {
    * 10% of the lockDuration value.
    * - **Default**: `2` seconds.
    */
-  receiveBatch(
+  async receiveBatch(
```

Comment:
Do we need to change the places where we are calling this?

Reply:
Its only being called in the test cases. And no the way it is called doesnt need to change. A function that returns a promise can be called with await even if it hasnt been marked as async

Reply:
The problem here was that since it wasnt marked as async, errors thrown from inside would only be caught in a try/catch

Marking it as async lets us attach a .catch() to the call

Reply:
If it helps, I often explain it thusly: from the outside, there is very little observable semantic difference between

```
async function foo () { ... }`
```

and

```
function foo() {
  return new Promise(() => { ... });
}
```

Users can await using async functions or call and use promise methods, either works fine. Further, it is not a breaking change to refactor a promise-returning function that looks like above into an async function.

In this case, the fix is to bring the code in line with either pattern. Making the function async is one way to do it. Another would be to move the `new Promise()` part to the top of the function so that the error checking stuff happens inside the promise constructor callback (so any errors thrown will reject the promise rather than bubble out synchronously).